### PR TITLE
PR for Issue #27

### PR DESCRIPTION
**Pull request by tesla-srt**
As per issue #27 - Add Right Mouse Button Support

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -29,6 +29,8 @@ async def websocket_endpoint(websocket: WebSocket):
                 pyautogui.moveRel(data['x'], data['y'])
             elif data['type'] == "tap":
                 pyautogui.leftClick()
+            elif data['type'] == "doubletap":
```
Comment (shubham1172) [inline main.py:32]: nit: should we make it `double_tap` for readability?

File: static/index.html [hunk 2]
```diff
@@ -42,7 +42,7 @@ <h1 class="title">Remote Play</h1>
       <div class="touchpad" id="touchpad">
         <p>
           Use this area as a touchpad.<br />Supported gestures:<br />track and
-          tap (Left Mouse Button).
+          tap (Left Mouse Button).<br />Double Tap (Right Mouse Button).
```
Comment (shubham1172) [inline static/index.html:45]: Let's break this into multiple lines for readability. Also, we should have a line break before `Double tap`, as that's also a supported gesture:
```html
Use this area as a touchpad.<br />
Supported gestures:<br />
Track, tap (left mouse button), and double tap (right mouse button).
```

File: package-lock.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (shubham1172) [inline package-lock.json:1]: Did you add this by mistake?

File: static/index.js [hunk 4]
```diff
@@ -22,19 +22,43 @@ function touchMove(ev) {
   const offsetX = ev.changedTouches[0].clientX - start.x;
   const offsetY = ev.changedTouches[0].clientY - start.y;
 
-  ws.send(JSON.stringify({ type: 'move', x: offsetX * 2, y: offsetY * 2 }));
+  ws.send(JSON.stringify({type: "move", x: offsetX * 2, y: offsetY * 2}));
```
Comment (shubham1172) [inline static/index.js:25]: nit: add spaces before and after curly braces.

File: static/index.js [hunk 5]
```diff
@@ -22,19 +22,43 @@ function touchMove(ev) {
...
-  const hammer = new Hammer(document.getElementById('touchpad'));
-  hammer.on('tap', (ev) => {
+  hammer.add([doubleTap, singleTap]);
+  doubleTap.recognizeWith(singleTap);
+
+  doubleTap.requireFailure(singleTap);
+  singleTap.requireFailure(doubleTap);
+
+  hammer.on("tap", ev => {
+    // left mouse button click
```
Comment (shubham1172) [inline static/index.js:57]: The one below should be 
```js
// right mouse button click
```

File: static/index.js [hunk 6]
```diff
@@ -22,19 +22,43 @@ function touchMove(ev) {
...
+    ws.send(JSON.stringify({type: "tap"}));
+  });
+  hammer.on("doubletap", ev => {
     // left mouse button click
-    ws.send(JSON.stringify({ type: 'tap' }));
+    ws.send(JSON.stringify({type: "doubletap"}));
```
Comment (shubham1172) [inline static/index.js:62]: `double_tap`

File: static/index.js [hunk 7]
```diff
@@ -22,19 +22,43 @@ function touchMove(ev) {
...
+  });
+  var DoubleTap = new Hammer.Tap({
+    event: "doubletap",
+    taps: 2
+  });
+  const hammer = new Hammer.Manager(document.getElementById("touchpad"));
+
+
+  var singleTap = new Hammer.Tap({ event: 'tap' });
+  var doubleTap = new Hammer.Tap({event: 'doubletap', taps: 2 });
```
Comment (shubham1172) [inline static/index.js:48]: nit: space after and before curly braces

File: static/index.js (hunk 6 above)
Comment (shubham1172) [inline static/index.js:62]: nit: space after and before curly braces

File: static/index.js [hunk 8]
```diff
@@ -22,19 +22,43 @@ function touchMove(ev) {
...
 
-document.addEventListener('DOMContentLoaded', (event) => {
-  document.getElementById('touchpad').addEventListener('touchstart', touchStart);
-  document.getElementById('touchpad').addEventListener('touchmove', touchMove);
+document.addEventListener("DOMContentLoaded", event => {
+  document
+    .getElementById("touchpad")
+    .addEventListener("touchstart", touchStart);
+  document.getElementById("touchpad").addEventListener("touchmove", touchMove);
+  var Tap = new Hammer.Tap({
```
Comment (shubham1172) [inline static/index.js:36]: Line 36 and 40 are extra. Same as line 47 and 48.

File: main.py (hunk 1 above)
Comment (shubham1172) [reply main.py:32]: Actually it's consistent with hammer, similar to `tap`, so let's let it be.

File: static/index.js [hunk 9]
```diff
@@ -22,19 +22,43 @@ function touchMove(ev) {
...
+  const hammer = new Hammer.Manager(document.getElementById("touchpad"));
+
+
+  var singleTap = new Hammer.Tap({ event: 'tap' });
+  var doubleTap = new Hammer.Tap({event: 'doubletap', taps: 2 });
 
-  const hammer = new Hammer(document.getElementById('touchpad'));
-  hammer.on('tap', (ev) => {
+  hammer.add([doubleTap, singleTap]);
+  doubleTap.recognizeWith(singleTap);
```
Comment (shubham1172) [inline static/index.js:51]: Do we want this? https://hammerjs.github.io/recognize-with/

Comment (shubham1172) [reply static/index.js:51]: BTW, you can use this for testing https://gutool.com/test-mouse/

**Comment by tesla-srt** [comment]
I see, okay will review your notes and integrate this evening. 
Seems my beautifier plugin didn't like the spaces. Will adjust.

File: main.py (hunk 1 above)
Comment (tesla-srt) [reply main.py:32]: I agree. Just went back and played around with the semantics and it just _looked off_ during reading.

File: package-lock.json (hunk 3 above)
Comment (tesla-srt) [reply package-lock.json]: Yes, not entirely sure how this got here.

File: static/index.js [hunk 10]
```diff
@@ -1,12 +1,12 @@
 const ws = new WebSocket(`ws://${window.location.host}/ws`);
-const start = { x: 0, y: 0 };
+const start = {x: 0, y: 0};
```
[the remark this answers is not in the document]
Comment (tesla-srt) [reply static/index.js]: Updated my Beautify settings. Should be good now, lmk.

File: static/index.js (hunk 9 above)
Comment (tesla-srt) [reply static/index.js:46]: I think if ever there's a need to add a middle mouse click then this way would make it straightforawrd

Comment (tesla-srt) [reply static/index.js:46]: Or perhaps, scrolling (two finger swipe?)

Comment (shubham1172) [reply static/index.js:46]: Yes, I think scrolling will be useful, I will create an issue for the same.

File: static/index.js [hunk 11]
```diff
@@ -22,19 +22,36 @@ function touchMove(ev) {
...
 }
 
-document.addEventListener('DOMContentLoaded', (event) => {
-  document.getElementById('touchpad').addEventListener('touchstart', touchStart);
-  document.getElementById('touchpad').addEventListener('touchmove', touchMove);
-
-  const hammer = new Hammer(document.getElementById('touchpad'));
-  hammer.on('tap', (ev) => {
+document.addEventListener("DOMContentLoaded", event => {
+  document
```
Comment (shubham1172) [inline static/index.js:32]: Should we not have a line break here?

Comment (shubham1172) [reply static/index.js:32]: *Consistency with the line below.

File: README.md [hunk 12]
```diff
@@ -12,6 +12,7 @@ The following actions are supported:
 - Seek controls (<< and >>)
 - Mouse cursor movement
 - Left-mouse-button click using tapping
+- Right-mouse-button click using double tap
```
Comment (shubham1172) [inline README.md:15]: Can you also please update the screenshot in this image to reflect the same?

File: static/index.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (shubham1172) [reply static/index.js]: Thanks!

File: static/index.js (hunk 11 above)
Comment (tesla-srt) [reply static/index.js:32]: Got you! Check latest commit

File: README.md (hunk 12 above)
Comment (tesla-srt) [reply README.md:15]: Will do :)

**Review by shubham1172 — APPROVED** [review]
LGTM, thanks for the PR @tesla-srt!

**Pull request merged**